documentation/guides/migrating-from-v3-to-v4.md

```diff
@@ -0,0 +1,417 @@
+    - [groups](#polaris-groups)
+  - [Icon](#polaris-icon)
+    - [source](#polaris-source)
+    - [untrusted](#polaris-untrusted)
+- [Removed Exports/Components](#polaris-removed-exportscomponents)
+  - [WithContext](#polaris-withcontext)
+  - [WithRef](#polaris-withref)
+  - [Autocomplete.ComboBox.TextField](#polaris-autocompletecomboboxtextfield)
+  - [Autocomplete.ComboBox.OptionList](#polaris-autocompletecomboboxoptionlist)
+  - [Navigation.UserMenu](#polaris-navigationusermenu)
+  - [Modal.Dialog/Tabs.Panel](#polaris-dialog-panel)
+- [Dependencies](#polaris-dependencies)
+- [Typescript](#polaris-typescript)
+  - [Config](#polaris-config)
+  - [Types](#polaris-types)
+
+## Testing <a name="polaris-testing"></a>
+
+We’ve migrated to [React’s new context API](https://reactjs.org/docs/context.html) while restructuring Polaris React’s entire context structure. Using the Polaris test provider will allow you to keep up to date on Polaris React's internal contexts.
+
+In v3, you could hook into Polaris's context types.
+
+**Note:** These examples use [Enzyme](https://airbnb.io/enzyme/). Other testing libraries can also be used, as shown in [these examples](https://github.com/Shopify/polaris-react/tree/master/examples).
+
+```jsx
+// old
+import merge from 'lodash/merge';
+import {shallow, mount} from 'enzyme';
+import {createPolarisContext, polarisContextTypes} from '@shopify/polaris';
+
+function mergeAppProviderOptions(options) {
+  const context = createPolarisContext();
+
+  return merge(
+    {},
+    {
+      context,
+      childContextTypes: polarisContextTypes,
+    },
+    options,
+  );
+}
+
+export function mountWithAppProvider(node, options) {
+  return mount(node, mergeAppProviderOptions(options));
+}
+
+export function shallowWithAppProvider(node, options) {
+  return shallow(node, mergeAppProviderOptions(options)).dive(options);
+}
+```
+
+Moving forward you'll want to wrap your code in the Polaris test provider.
+
+```jsx
+// new
+import {mount} from 'enzyme';
+import {PolarisTestProvider} from '@shopify/polaris';
+
+export function mountWithPolaris(node) {
+  return mount<P>(
+    <PolarisTestProvider {...context}>{node}</PolarisTestProvider>,
+  );
+}
+```
+
+## Component API changes <a name="polaris-component-api-changes"></a>
+
+### AppProvider <a name="polaris-appprovider"></a>
+
+#### i18n <a name="polaris-i18n"></a>
+
+i18n is now a required prop and Polaris includes [many translations](https://github.com/Shopify/polaris-react/blob/master/locales) to support internationalization.
+
+```jsx
+import fr from '@shopify/polaris/locales/fr.json';
+...
+<AppProvider i18n={fr}>
+...
+```
+
+### Navigation <a name="polaris-navigation"></a>
+
+#### iconBody <a name="polaris-icon-body"></a>
+
+<!-- ping ben or andre about this <svg><path d='M17 9h-6V3a1 1 0 1 0-2 0v6H3a1 1 0 1 0 0 2h6v6a1 1 0 1 0 2 0v-6h6a1 1 0 1 0 0-2'  fill-rule='evenodd'/></svg> doesnt seem to work -->
+
+Pass a string to the icon prop instead.
```

Comment:
`iconBody` is now deprecated. Pass a string to the icon prop instead.

Reply:
```suggestion
Pass a string to the icon prop instead.

```jsx
 const navigationSection = (
    <Navigation.Section
      items={[
        {
          url: '/path/to/place',
          label: 'Products',
          icon: "<svg viewBox='0 0 20 20' xmlns='http://www.w3.org/2000/svg'><path d='M10.707 17.707l5-5a.999.999 0 1 0-1.414-1.414L11 14.586V3a1 1 0 1 0-2 0v11.586l-3.293-3.293a.999.999 0 1 0-1.414 1.414l5 5a.999.999 0 0 0 1.414 0' /></svg>",
        },
        {
          url: '/path/to/place',
          label: 'Orders',
          icon: OrdersMajorTwotone,
          badge: '15',
        }
      ]}
    />
  );
  ```
```